core/src/main/java/com/tickaroo/tikxml/XmlReader.java

```diff
@@ -313,7 +313,7 @@ private int doPeek() throws IOException {
    * @throws IOException
    */
   private boolean isCDATA() throws IOException {
-    return buffer.rangeEquals(0, CDATA_OPEN);
+    return fillBuffer(CDATA_OPEN.size()) && buffer.rangeEquals(0, CDATA_OPEN);
```

Comment:
In case this is still keeping you up at night, I think this is related to the magic number 2 from this PR.
https://github.com/Tickaroo/tikxml/pull/133#discussion_r365294682
It used to be four, for the `CDATA`. Two accommodates for an opening comment `<-`.

Reply:
ahaha... thanks, figured that out in the meantime